addons/binding/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/discovery/EvohomeDiscoveryService.java

```diff
@@ -0,0 +1,130 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EvohomeDiscoveryService} class is capable of discovering the available data from Evohome
+ *
+ * @author Neil Renaud - Initial contribution
+ */
+public class EvohomeDiscoveryService extends AbstractDiscoveryService {
+    private Logger logger = LoggerFactory.getLogger(EvohomeDiscoveryService.class);
+    private static final int SEARCH_TIME = 2;
+
+    private EvohomeGatewayHandler evohomeGatewayHandler;
+
+    public EvohomeDiscoveryService(EvohomeGatewayHandler evohomeBridgeHandler) {
+        super(EvohomeBindingConstants.SUPPORTED_THING_TYPES_UIDS, SEARCH_TIME);
+        this.evohomeGatewayHandler = evohomeBridgeHandler;
+    }
+
+    @Override
+    public void startScan() {
+        logger.debug("Evohome start scan");
+
+        if (evohomeGatewayHandler != null) {
+            try {
+                EvohomeApiClient client = evohomeGatewayHandler.getApiClient();
+                if (client != null) {
+                    for (ControlSystem controlSystem : client.getControlSystems()) {
+                        discoverDisplay(controlSystem);
+                        discoverHeatingZones(controlSystem.getId(), controlSystem.getHeatingZones());
+                    }
+                }
+            } catch (Exception e) {
+                logger.warn("{}", e.getMessage(), e);
+            }
+        }
+
+        stopScan();
+    }
+
+    private void discoverHeatingZones(String locationId, TemperatureControlSystem heatingZones) {
+        for(Zone zone : heatingZones.zones){
+            String zoneName = zone.name;
+            String zoneId = zone.zoneId;
+            String modelType = zone.modelType;
+            String zoneType = zone.zoneType;
+            HeatSetpointCapabilities heatSetpointCapabilities = zone.heatSetpointCapabilities;
+            ScheduleCapabilities scheduleCapabilities = zone.scheduleCapabilities;
+            ThingUID thingUID = findThingUID(EvohomeBindingConstants.THING_TYPE_EVOHOME_HEATING_ZONE.getId(), zoneName);
+            Map<String, Object> properties = new HashMap<>();
+            properties.put(EvohomeBindingConstants.LOCATION_ID, locationId);
+            properties.put(EvohomeBindingConstants.ZONE_ID, zoneId);
+            properties.put(EvohomeBindingConstants.ZONE_NAME, zoneName);
+            properties.put(EvohomeBindingConstants.ZONE_TYPE, zoneType);
+            properties.put(EvohomeBindingConstants.ZONE_MODEL_TYPE, modelType);
+            addDiscoveredThing(thingUID, properties, zoneName);
+        }
+    }
+
+    private void discoverDisplay(ControlSystem controlSystem) {
+        String name = controlSystem.getName();
+        ThingUID thingUID = findThingUID(EvohomeBindingConstants.THING_TYPE_EVOHOME_DISPLAY.getId(), name);
+        Map<String, Object> properties = new HashMap<>();
+
+        //TODO NR: Doesn't the gateway also have a location? I don't have a dual location system but from the EvoHome UI
```

Comment:
The TODO basically dictates a test with multiple locations. I don't know anyone with such a system, so testing it would be rather difficult. Would it be ok to put this in the docs? Currently the evohome binding only supports one location?